src/utils.rs

```diff
@@ -55,4 +66,43 @@ mod tests {
             vec!["This", " is", " a", " test", "        ", " with", " a", " lot", " of", " spaces"]
         );
     }
+
+    // Mapping % target to temperature range
+    #[test]
+    fn test_target_at_min() {
+        assert_eq!(map_to_range(0, 100, 0), 0.0);
+        assert_eq!(map_to_range(10, 20, 0), 10.0);
+    }
+
+    #[test]
+    fn test_target_at_max() {
+        assert_eq!(map_to_range(0, 100, 100), 100.0);
+        assert_eq!(map_to_range(10, 20, 100), 20.0);
+    }
+
+    #[test]
+    fn test_target_in_middle() {
+        assert_eq!(map_to_range(0, 100, 50), 50.0);
+        assert_eq!(map_to_range(10, 20, 50), 15.0);
+        assert_eq!(map_to_range(0, 1, 50), 0.5);
+    }
+
+    #[test]
+    fn test_target_out_of_bounds() {
+        assert_eq!(map_to_range(0, 100, 3000), 100.0); // Cap to 100
```

Comment:
I would throw an error if temp is outside of range, else users might feel that temp is being set correctly, what do you think?

Reply:
Check out the implementation of `utils:: map_to_range`" 
```
// Cap the target to the percentage range [0, 100]
let capped_target = target.min(100);

// Calculate the target value in the range [min, max]
let range = max as f32 - min as f32;
let percentage = capped_target as f32 / 100.0;
min as f32 + (range * percentage)
```
It actually caps the input to 100. So it ensures that we always set the temp correctly. It is guaranteed to be within the min-max range.